Caches/ResponsesCache.cs

```diff
@@ -1,28 +1,112 @@
 using System;
 using System.Collections.Concurrent;
+using System.Linq;
 using System.Threading.Tasks;
 using Concept.Options;
 using Microsoft.Extensions.Options;
+using Theory.Infos;
+using Theory.Providers;
+using Theory.Search;
 
 namespace Concept.Caches
 {
     public sealed class ResponsesCache
     {
         private readonly CacheOptions _cacheOptions;
-        private readonly ConcurrentDictionary<object, object> _responses;
+        private readonly ConcurrentDictionary<DateTimeOffset, SearchResponse> _ytCache;
+        private readonly ConcurrentDictionary<DateTimeOffset, SearchResponse> _scCache;
+        private readonly ConcurrentDictionary<DateTimeOffset, SearchResponse> _bcCache;
 
         public ResponsesCache(IOptions<ApplicationOptions> aOptions)
         {
             _cacheOptions = aOptions.Value.CacheOptions;
-            _responses = new ConcurrentDictionary<object, object>();
+            _ytCache = new ConcurrentDictionary<DateTimeOffset, SearchResponse>();
+            _scCache = new ConcurrentDictionary<DateTimeOffset, SearchResponse>();
+            _bcCache = new ConcurrentDictionary<DateTimeOffset, SearchResponse>();
         }
 
         public async Task AutoPurgeAsync()
         {
-            while (_cacheOptions.IsEnabled)
+            while (_cacheOptions.IsEnabled && _cacheOptions.ExpiresAfter > 0)
```

Comment:
I think we should allow a minimum expire time. For example if it's below 30 minutes, we throw an exception.